Tools/Scripts/layout-test-worker-parser

```diff
@@ -237,22 +251,74 @@ def load_from_file(path):
 
 # Loads the stdio input for analysis from a build URL.
 # Returns a str with the contents of the stdio log.
-def load_from_buildbot(url):
-    url = url.replace('/#/', '/api/v2/') + ("/raw" if "/logs/stdio" in url else "/steps/layout-test/logs/stdio/raw")
-    try: # validation
-        uobj = urlparse(url)
+def load_from_url(base_url: str, step_name: str = None, no_warn: bool = False) -> str:
+    '''Takes a build URL and analyzes the steps to determine which log to return. Can also take a step stdio link ("/steps/<step-name>/logs/stdio") or a .txt file URL.'''
+
+    if not base_url.endswith('.txt'):
+        base_url = base_url.replace('/#/', '/api/v2/').strip('/')
+        # Don't analyze steps if user specifically asked for stdio
+        if '/logs/stdio' not in base_url:
+            base_url += '/steps'
+        elif not base_url.endswith('/raw'):
+            base_url += '/raw'
+
+    try:
+        uobj = urlparse(base_url)
         if not all([uobj.scheme, uobj.netloc]):
             raise IOError()
     except:
-        error("Invalid URL specified: %s.", url)
+        error(f'Invalid URL specified: {base_url}.')
+        graceful_exit(os.EX_DATAERR)
+
+    info('Loading from URL (this may take a moment)...')
+
+    try:
+        data = requests.get(base_url)
+    except Exception as e:
+        error(f'Unable to load data: {e.args[0]}. Check your spelling and try again.')
         graceful_exit(os.EX_DATAERR)
 
-    info("Loading from URL (this may take a moment)...")
+    # Early return for stdio/txt result
+    if base_url.endswith('/raw') or base_url.endswith('.txt'):
+        return data.text
+
+    # Analyze steps
+    candidate_steps = {}
+    layout_test_step = {}
+    build_data = json.loads(data.text)
+    for step in build_data['steps']:
+        if step['name'] == 'layout-test':
```

Comment:
Seems like this should be a regex rather than a bunch of `if` statements.

Reply:
I'm not sure a regex is necessary here. Since each of the `if` statements check distinct, non-abstract conditions, a regex seems like overkill.